packages/react-liveness/src/components/FaceLivenessDetector/service/machine/machine.ts

```diff
@@ -200,24 +200,57 @@ export const livenessMachine = createMachine<LivenessContext, LivenessEvent>(
       },
     },
     states: {
-      cameraCheck: {
-        entry: 'resetErrorState',
-        invoke: {
-          src: 'checkVirtualCameraAndGetStream',
-          onDone: {
-            target: 'waitForDOMAndCameraDetails',
-            actions: 'updateVideoMediaStream',
+      initCamera: {
+        initial: 'cameraCheck',
+        states: {
+          cameraCheck: {
+            entry: 'resetErrorState',
+            invoke: {
+              src: 'checkVirtualCameraAndGetStream',
+              onDone: {
+                target: 'waitForDOMAndCameraDetails',
+                actions: 'updateVideoMediaStream',
+              },
+              onError: {
+                target: '#livenessMachine.permissionDenied',
+              },
+            },
+          },
+          waitForDOMAndCameraDetails: {
+            after: {
+              0: {
+                target: '#livenessMachine.initWebsocket',
+                cond: 'hasDOMAndCameraDetails',
+              },
+              10: { target: 'waitForDOMAndCameraDetails' },
+            },
           },
-          onError: { target: 'permissionDenied' },
         },
       },
-      waitForDOMAndCameraDetails: {
-        after: {
-          0: {
-            cond: 'hasDOMAndCameraDetails',
-            target: 'start',
+      initWebsocket: {
+        initial: 'initializeLivenessStream',
+        states: {
+          initializeLivenessStream: {
+            invoke: {
+              src: 'openLivenessStreamConnection',
+              onDone: {
+                target: 'waitForSessionInfo',
+                actions: [
+                  'updateLivenessStreamProvider',
+                  'spawnResponseStreamActor',
+                ],
+              },
+            },
+          },
+          waitForSessionInfo: {
+            after: {
+              0: {
+                target: '#livenessMachine.start',
+                cond: 'hasServerSessionInfo',
+              },
+              100: { target: 'waitForSessionInfo' },
+            },
```

Comment:
Does this need to poll? Wondering if we can just treat it as a service call

Reply:
If we were to change this to a service call, wouldnt somewhere down the line we need to poll for the event to come in? I can't currently think of a way that we can make an function that will await until we receive an event on the websocket connection, unless that function involved a poll

Reply:
Think it can be done by refactoring the chain of state updates related to the `SET_SESSION_INFO` event, but that much of a refactor but might equate to scope creep for this PR